types/DocumentTypes.ts

```diff
@@ -68,6 +68,9 @@ export enum ImageFrameSourceType {
     assetProvider = 'assetProvider',
     variable = 'variable',
 }
+
+export type ImageSource = ImageFrameSource | UrlImageFrameSource | null;
```

Comment:
It already exists in `frameTypes.ts` as `ImageFrameSource`.

`export type ImageFrameSource = ImageUrlVariableSource | ImageUrlSource | ImageConnectorVariableSource;`

I noted that It is missing `ImageAssetProviderSource`.

also looking at `ImageConnectorVariableSource`, `variableId` property is missing as well.

In any case, if you are wondering why we have subtypes of variable in there it's probably because we shouldn't. I made a ticket for it https://support.chili-publish.com/browse/EDT-758

I vote to align on that one when you're all back from holidays 🎄.

Reply:
I've noticed there is a `url` property in `ImageConnectorVariableSource`. Not sure if it should be there